OpenImageIOReader : Disable channelName heuristics for single-part files

danieldresser-ie opened the pull request:
As discussed here: https://github.com/GafferHQ/gaffer/issues/6527 and in various other conversations, it's rather unclear what do to with single part files with a part name metadata stored in them.

Nuke doesn't intentionally produce these files, but it's possible in Nuke to accidentally to create a file with garbage stored in this metadata, and Nuke will ignore it on load.

We're not sure whether there might be some hypothetical package that also produces non-standards-compliant exrs that would benefit from the current behaviour, but changing the behaviour to match Nuke is pretty straightforward, this commit should do it.

If we decide to go with this approach, it needs a little bit of documentation and a test, but I figured I'd throw this up in it's current form for discussion.

johnhaddon reviewed (commented):
Thanks Daniel. Still not sure how I feel about this one exactly, but I've made a performance-related comment inline that I think is worth doing at least a little bit of testing around if we're going to follow through on this...

danieldresser-ie commented:
The final conclusion of our various offline conversations is that we are more concerned with file with bogus part names output accidentally from Nuke ( which we know does happen in practice ), than we are concerned with hypothetical software that might intentionally use the part name in single part files ( The only software we've found that is sort of set up to use the part name in single part files is Silhouette, but it specifically sets the part name to something useless when writing single part files ).

That means we should go ahead and take this approach.

I've switched to using oiio:subimages, added a basic test, and added a Changelog entry.
